services/scabbard/src/cli/key.rs

```diff
@@ -0,0 +1,200 @@
+// Copyright 2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::PathBuf;
+
+use sawtooth_sdk::signing::secp256k1::Secp256k1PrivateKey;
+
+use super::error::CliError;
+
+/// Load a private key from the local filesystem.
+///
+/// If the argument is a file path (contains a '/'), this will attempt to load the key file from
+/// the specified location. If the argument is not a file path, this will attempt to load the
+/// file from the $HOME/.scabbard/keys directory; when loading from this directory, the '.prive'
+/// file extension is optional.
+pub fn load_signing_key(key: &str) -> Result<Secp256k1PrivateKey, CliError> {
+    let file_path = determine_key_file_path(key)?;
+
+    let key_file = File::open(file_path).map_err(|err| {
+        CliError::action_error_with_source("failed to open private key file", err.into())
+    })?;
+    let mut key_file_reader = BufReader::new(key_file);
+
+    let mut raw_key_string = String::new();
+    key_file_reader
+        .read_line(&mut raw_key_string)
+        .map_err(|err| {
+            CliError::action_error_with_source("failed to read private key file", err.into())
+        })?;
+
+    let key_string = raw_key_string.trim();
+    if key_string.is_empty() {
+        return Err(CliError::action_error("private key file is empty"));
+    }
+
+    let signing_key = Secp256k1PrivateKey::from_hex(key_string).map_err(|err| {
+        CliError::action_error_with_source("failed to read valid private key from file", err.into())
+    })?;
+
+    Ok(signing_key)
+}
+
+fn determine_key_file_path(key: &str) -> Result<PathBuf, CliError> {
+    if key.contains('/') {
+        Ok(PathBuf::from(key))
+    } else {
+        let mut path = dirs::home_dir().ok_or_else(|| {
+            CliError::action_error("failed to load signing key: unable to determine home directory")
+        })?;
+        path.push(".scabbard");
```

Comment:
We should make this use .splinter (keys for the project, as opposed to keys for a particular CLI).

Reply:
Updated